doc: Secure boot doc update (#11540)
nrfconnect/sdk-nrf: doc/nrf/config_and_build/bootloaders_and_dfu/bootloader.rst

   :local:
   :depth: 2

> The |NCS|| (NCS) secure bootloader solutions are built on the *chain of trust* concept.

Reviewer: ```suggestion
The |NCS| (NCS) secure bootloader solutions are built on the *chain of trust* concept.
```

> The nRF Connect SDK (NCS) secure bootloader solutions are built on the *chain of trust* concept.

Reviewer 2: ```suggestion
The |NCS| (NCS) secure bootloader solutions are built on the *chain of trust* concept.
```

Architecture
************

> The nRF Connect SDK (NCS) currently supports two implementations:

Reviewer 2: ```suggestion
The |NCS| currently supports two implementations:
```

This way, the immutable bootloader can guarantee that the next image in the boot sequence will not start up if it has been tampered with in any way.
For example, if an attacker attempts to take over the device by altering the firmware, the device will not boot, and thus not run the infected code.

> The immutable bootloader is locked to the flash memory and can't be modified or deleted without erasing the entire device.

Reviewer 2: ```suggestion
The immutable bootloader is locked to the flash memory and cannot be modified or deleted without erasing the entire device.
```

When programming an upgradable bootloader, the build system can automatically generate pre-signed variants of the image verified by the |NSIB|.

> The upgradable bootloader does not use pre-signed variants for updating the application.

Reviewer 2: ```suggestion
The upgradable bootloader does not use pre-signed variants to update the application.
```

> When building upgrade images for the image following the |NSIB| in the boot chain, like the upgradable bootloader or the application, you must build with pre-signed variants.

Reviewer 2: ```suggestion
When building upgrade images for the image following the |NSIB| in the boot chain, like the upgradable bootloader or application, you must build with pre-signed variants.
```
